### PR TITLE
[DO-4673] (minor) new test data set for Tier-2 supplier.

### PR DESCRIPTION
**Pull request by martin-fritz**
Additional set of test data. See description in https://cofinity-x.atlassian.net/browse/DO-4673.

### CODE REVIEW

**Comment by hgo-ds** [comment]
Please compare with #76, may lead to conflicts.

**Review by hgo-ds — COMMENTED** [review]
**Suggestion:**

- drop folder local in favor to folder testdat-a
- rename testdata-a and testdata-b to a more clear description about what data will b created, i.e. containing the BPN
- change the names of the corresponding application-yaml files in the same way

File: tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- this script is executed by flyway as part of a lifecycle hook after all migrations have been applied
+-- (see https://documentation.red-gate.com/fd/callback-concept-184127466.html).
+-- it is only intended for usage in local or test environments
+
+insert into assets_as_built_childs
+    (asset_as_built_id  , id                 , id_short)
+values
+    -- supplier is child of owner
+    (${assetAsBuiltId07}, ${assetAsBuiltId01}, 'H-LeftHeadLight'),  -- Z1 isParentOf Xenon Left-Headlights
+    (${assetAsBuiltId07}, ${assetAsBuiltId02}, 'H-RightHeadLight'); -- Z1 isParentOf Xenon Right-Headlights
```
Comment (hgo-ds) [inline tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql:10]: **Clarification:**

I guess this needs to be turned around:
```suggestion
    (${assetAsBuiltId01}, ${assetAsBuiltId07}, 'H-LeftHeadLight'),  -- Xenon Left-Headlights isChildOf Z1
    (${assetAsBuiltId02}, ${assetAsBuiltId07}, 'H-RightHeadLight'); -- Xenon Right-Headlights isChildOf Z1
```

File: tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__012_asset_as_built_parents.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- this script is executed by flyway as part of a lifecycle hook after all migrations have been applied
+-- (see https://documentation.red-gate.com/fd/callback-concept-184127466.html).
+-- it is only intended for usage in local or test environments
+
+insert into assets_as_built_parents
+    (asset_as_built_id  , id                 , id_short)
+values
+    -- owner is parent of supplier
+    (${assetAsBuiltId01}, ${assetAsBuiltId07}, 'BMW-Z1'),           -- Xenon Left-Headlights isChildOf Z1
+    (${assetAsBuiltId02}, ${assetAsBuiltId07}, 'BMW-Z1');           -- Xenon Right-Headlights isChildOf Z1
```
Comment (hgo-ds) [inline tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__012_asset_as_built_parents.sql:10]: **Clarification:**

I guess this needs to be turned around:
```suggestion
    (${assetAsBuiltId07}, ${assetAsBuiltId01}, 'BMW-Z1'), --  Z1 isParentOf Xenon Left-Headlights
    (${assetAsBuiltId07}, ${assetAsBuiltId02}, 'BMW-Z1'); --  Z1 isParentOf Xenon Right-Headlights

```

**Review by hgo-ds — COMMENTED** [review]
Left a suggestion in the ticket about possible extension for truncate profile to clean db on purpose.

File: tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql (hunk 1 above)
Comment (hgo-ds) [inline tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql:10]: **Suggestion:**
adjust comments:
```suggestion
    -- owner has child supplier
    (${assetAsBuiltId07}, ${assetAsBuiltId01}, 'H-LeftHeadLight'),  -- Z1 has child Xenon Left-Headlights
    (${assetAsBuiltId07}, ${assetAsBuiltId02}, 'H-RightHeadLight'); -- Z1 has child Xenon Right-Headlights
```

File: tx-backend/INSTALL.md [hunk 3]
```diff
@@ -63,7 +63,7 @@ spring:
 ```
 Database scripts are executed with Flyway. Put the scripts at [migration](src/main/resources/db/migration).
 
-Testdata are created by activating profile 'testdata', scripts are located at [testdata](src/main/resources/db/testdata).
+Testdata are created by activating profile 'testdata', scripts are located at [testdata](src/main/resources/db/local).
```
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/INSTALL.md]: done

File: tx-backend/src/main/resources/application-testdata-tier2.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
```
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/src/main/resources/application-testdata-tier2.yml]: Application BPN should be set via the environment profiles (dev, int, etc.), not via the testdata profiles.

File: tx-backend/src/main/resources/application-testdata-b.yml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
```
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/src/main/resources/application-testdata-b.yml]: see above.

File: tx-backend/src/main/resources/application-testdata-osram.yml [hunk 6]
```diff
@@ -21,11 +21,11 @@
```
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/src/main/resources/application-testdata-osram.yml:21]: See above.

File: tx-backend/src/main/resources/application-testdata-bmw.yml [hunk 7]
```diff
@@ -21,7 +21,7 @@
```
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/src/main/resources/application-testdata-bmw.yml:21]: See above.

File: tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__012_asset_as_built_parents.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (martin-fritz) [reply tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__012_asset_as_built_parents.sql:10]: fixed.

File: tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql (hunk 1 above)
Comment (martin-fritz) [reply tx-backend/src/main/resources/db/testdata-bmw/afterMigrate__011_asset_as_built_childs.sql:10]: fixed.

**Review by hgo-ds — APPROVED** [review]
LGTM

We should collect the data issues to the mentioned ticket [DO-4702](https://cofinity-x.atlassian.net/browse/DO-4702)

**Comment by hgo-ds** [comment]
fix of remaining issues is covered by https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/pull/180

**Pull request merged**